CRM/Contribute/BAO/ContributionSoft.php

```diff
@@ -236,11 +236,14 @@ public static function retrieve(&$params, &$defaults) {
   public static function del($params) {
     //delete from contribution soft table
     $contributionSoft = new CRM_Contribute_DAO_ContributionSoft();
-    $contributionSoft->id = CRM_Utils_Array::value('id', $params);
-    if (!$contributionSoft->find()) {
-      return FALSE;
+    if (!empty($params['id'])) {
+      $contributionSoft->id = CRM_Utils_Array::value('id', $params);
+      if (!$contributionSoft->find()) {
+        return FALSE;
+      }
+      // Are we really sure about this??
```

Comment:
Good question. My guess is that this is to handle case where the soft contributions are successfully deleted and we don't want the url to cause an error by referring to something just deleted. However, the deletion might fail, in which case the url should go back to same soft contribution. Could you move the unset to between L250-251 ?